[TIMOB-24963] iOS/Android: Add scrollToTop() to ScrollView

m1ga opened the pull request:
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24963

sgtcoolguy committed:
Use v8 inspector protocol for debugger
- Add java websocket library so we can use websockets to talk to the inspector/debugger.
- Use V8 5.8.283.32

sgtcoolguy committed:
Fix double-definition of getCenter()

sgtcoolguy committed:
Bump to V8 5.9.211.33

hansemannn reviewed (approved):
LGTM. Will await the Android review as well.

hansemannn committed:
[TIMOB-25005] Remove Ti.UI.iPhone namespace

hansemannn committed:
[TIMOB-25005] Address review comments

Commit:
[TIMOB-10724] Remove Android module gperf dependency

sgtcoolguy committed:
Bump to V8 5.9.211.41

lokeshchdhry commented:
@hansemannn , @garymathews, Is the android CR done for this ?

Commit:
[TIMOB-10724] Use FNV-1a hash algorithm

hansemannn commented:
@lokeshchdhry Not, yet. Awaiting the CR from @garymathews, but iOS is approved.

hansemannn commented:
@m1ga As 6.2.0 is branched already and this is a minor feature, please change the `since` to `7.0.0`, thx!

maggieaxway committed:
[TIMOB-6392]- Android fix: Ti.UI.Button text values with too many displayable characters draw improperly

[TIMOB-6392]- Android fix: Ti.UI.Button text values with too many
displayable characters draw improperly

maggieaxway committed:
Update TiUIButton.java

For parity with iOS:

Force to set single line
Use TextUtils.TruncateAt.MIDDLE ellipsis mode

maggieaxway committed:
Update TiUIButton.java

Move changes to constructor to make sure always applied

hansemannn committed:
[TIMOB-25005] Remove old files

hansemannn committed:
[TIMOB-25005] Fix doc-validation issues

feons committed:
[CLI-649] direct users to use appc ti config if run from appc

sgtcoolguy committed:
Use V8 6.0.286.55 to build

sgtcoolguy committed:
Use android NDK r12b to build, use ANDROID_SDK env var to point to install in Jenkinsfile.

sgtcoolguy committed:
Instead of wrapping app.js in function for debug pause on start, pass the source and filename as-is and let the inspector compile the source, schedule a pause and then run the script. Avoids the additional injected wrapping function.

sgtcoolguy committed:
Begin hooking up eslint to the core javascript inside Android 'kernel'.

ewanharris committed:
[TIMOB-25216] Include Windows Modules docs

sgtcoolguy committed:
Bump to V8 6.1.534.33

Commit:
[TIMOB-25323] Release V8Object when KrollRuntime is alive

Commit:
[TIMOB-25324] Remove WeakReference from AssetManager

Commit:
[TIMOB-25353] Allow Titanium modules to process module dependencies

Commit:
[TIMOB-25353] Use appc-aar-tools 1.1.4

Commit:
[TIMOB-25359] Fix Ti.UI.ScrollView when keyboard is active

hansemannn committed:
[TIMOB-25346] Restore SBJSON for encoded strings

sgtcoolguy committed:
Revert "Update modules.json"

This reverts commit ae8343301d5e4582c9c0a4499f792e60169eb6a2.

Commit:
Merge branch 'master' into TIMOB-25238

Commit:
[TIMOB-25363] Use ContentProvider for Intent DATA

Commit:
Merge pull request #9410 from garymathews/TIMOB-25238

[TIMOB-25238] Android: Refactor TiBorderWrapperView to improve performance

Commit:
Merge branch 'master' into TIMOB-25266

Commit:
Merge pull request #9432 from garymathews/TIMOB-25266

[TIMOB-25266] Android: Fix WebView HW-acceleration for Android 4.1

garymathews committed:
Merge branch 'master' into TIMOB-25051

Commit:
[TIMOB-25359] Missing imports

garymathews committed:
Merge branch 'master' into TIMOB-25359

sgtcoolguy committed:
No longer upload NSP results to Threadfix, the server seems to randomly stop accepting NSP uploads and cause build failures for us mysteriously.

Commit:
Merge branch 'master' into TIMOB-25359

vijaysingh-axway committed:
[TIMOB-25301]: iPhone X Need to be able to control the Insets / Layout Margins

Commit:
Merge pull request #9502 from garymathews/TIMOB-25359

[TIMOB-25359] Android: Fix Ti.UI.ScrollView when keyboard is active

vijaysingh-axway committed:
[TIMOB-25301]: window property passed to safeAreaView

vijaysingh-axway committed:
[TIMOB-25301]: Safe area layout calculated early

gimdongwoo committed:
add description, validation

gimdongwoo committed:
Merge branch 'master' into sigalg

gimdongwoo committed:
add default value for `sigalg`

gimdongwoo committed:
change indent

vijaysingh-axway committed:
[TIMOB-25301]:Indentation fixed

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-25301

ewanharris committed:
[TIMOB-25202] Update to windowlib@0.6.0

sgtcoolguy committed:
Add AppSpider config file

gimdongwoo committed:
Merge branch 'master' into sigalg

gimdongwoo committed:
remove deault value #9484

ypbnv committed:
Set default borderWidth value.
Change docs.

cb1kenobi committed:
Merge pull request #9484 from gimdongwoo/sigalg

[TIMOB-18441] Add `--sigalg` option on cli for overriding keystore signing algorithm.

jquick-axway committed:
[TIMOB-25360] Android: ScrollView enhancements

- [TIMOB-25197] Added RefreshControl support to vertical and horizontal ScrollViews.
- [TIMOB-8898] Added Ti.UI.FIll and Ti.UI.SIZE support to ScrollView "contentWidth" and "contentHeight" properties.
- [TIMOB-25360] ScrollView child views that use Ti.UI.FILL should do a 100% fill like iOS if contentWidth/Height properties are not set.
- [TIMOB-25377] ScrollView child percent size/position should be relative to container size, like iOS.
  * Was relative to ScrollView's scrollable content area.
- [TIMOB-12153] ListView/TableView height set to Ti.UI.SIZE will return wrong height when put inside a ScrollView.
  * Was returning a height equivalent to Ti.UI.FILL, which was limited to size of parent view or window.
- Improved ScrollView measurement/layout performance:
  * Was always doing a 2-pass measurement, which would cause issues with percentage calculations.
  * Now only does a 1-pass mearurement.

jquick-axway committed:
Merge branch 'master' into TIMOB-25360

vijaysingh-axway committed:
[TIMOB-25301] : early calculation of safe area for tab group’s window

gaspardle committed:
DrawerLayoutProxy: fix CalledFromWrongThreadException exceptions

jquick-axway committed:
[TIMOB-25376] Android: Fixed bug where scrolled in ListView rows forget their width/height settings.

- This was a pre-existing issue before 6.2.0, but the layout pin fill fixes made in 6.2.0 revealed it.
- Caused by ListView calling TiUIView.processProperties() with only changed properties. Since width/height did not change, they're null. This causes processProperties() to reset LayoutParams via TiConvert.fillLayout() to do pin-fillls instead.

vijaysingh-axway committed:
[TIMOB-25301]: Added safeAreaViewProxy after creating it

vijaysingh-axway committed:
[TIMOB-25301] : Support for split window

vijaysingh-axway committed:
[TIMOB-25358]:iOS Problem with Searchbar, SDK 6.2.0 and later (#9495)

* [TIMOB-25287] : iPhone X Ti.UI.SearchBar in list-view get wrongly positioned

* [TIMOB-25287]: xcode 9 check applied

* [TIMOB-25287]: Fix for TiUITableView

* [TIMOB-25358] : iOS Problem with Searchbar, SDK 6.2.0 and later

* [TIMOB-25358]: Origin calculation made simpler

* [TIMOB-25358]:searchView height set 56 for ios 11

* [TIMOB-25358]: searchview height set 56 for ios 11

* [TIMOB-25287] : iPhone X Ti.UI.SearchBar in list-view get wrongly positioned

* [TIMOB-25358] : Constant used for search bar height

* [TIMOB-25287] : Indentation fixed.

* [TIMOB-25358]: Indentation fixed

hansemannn committed:
Fix formatting

sgtcoolguy committed:
Replace nsp scan with dependency check, upload to Threadfix

vijaysingh-axway committed:
[TIMOB-25373] : iOS - Search result view of List did not close when w idow containing list view get closed (#9508)

* [TIMOB-25373] : iOS - Search result view of List did not close when widow containing list view get closed

* [TIMOB-25373] : Set searchresultupdater to nil in dealloc

hansemannn committed:
[TIMOB-24805] iOS11: Support large window titles (#9136)

* [TIMOB-24805] Support large window titles

* [TIMOB-24805] Add missing docs, update build settings

* [TIMOB-24805] Guard setter again

* [TIMOB-24805] Address review-comments

* [TIMOB-24805] Fix clang-format issue

* [TIMOB-24805] Wrap API's different

# Conflicts:
#	iphone/Classes/TiUIiOSProxy.h

* [TIMOB-24805] Fix guard-macros

* [TIMOB-24805] Change "since" to 6.3.0

* [TIMOB-24805] Add titleAttributes for large titles

* [TIMOB-24805] Apply titleAttributes to both titles

hansemannn committed:
Merge branch 'master' into TIMOB-25301

jquick-axway committed:
Android: Code changes based on feedback from [TIMOB-25360] commit.

jquick-axway committed:
Merge branch 'master' into TIMOB-25360

Commit:
[TIMOB-25382] Fix MSG IDs and formatting

hansemannn committed:
Rephrase safe-area docs

hansemannn committed:
Update Window.yml

hansemannn committed:
[TIMOB-25300] Fix title-control in landscape

hansemannn committed:
Fix launch-screen size

hansemannn committed:
Merge branch 'master' into TIMOB-25300

ssjsamir committed:
Merge branch 'master' into TIMOB-25376

ypbnv committed:
Fix window flags for extendBackground.

infosia committed:
[TIMOB-25400] Windows: Update Windows module apiversion

janvennemann committed:
[TIMOB-25393] Handle rpath search paths as string (#9528)

Switches the LD_RUNPATH_SEARCH_PATH build setting to be handled as a space separated string of paths instead of an array of paths. Xcode can handle both, but the string representation is the default one for this build setting so we stick to that.

hansemannn committed:
Update modules.json

garymathews committed:
Merge branch 'master' into TIMOB-25363

jquick-axway committed:
Merge branch 'master' into TIMOB-25376

ewanharris committed:
Include note about required naming convention (#9511)

jquick-axway committed:
Merge branch 'master' into TIMOB-25360

jquick-axway committed:
Merge branch 'master' into TIMOB-25376

garymathews committed:
Merge branch 'master' into TIMOB-25363

garymathews committed:
Merge pull request #9500 from garymathews/TIMOB-25363

[TIMOB-25363] Android: Use ContentProvider for Intent data

Commit:
Merge branch 'master' into TIMOB-25376

Commit:
Merge pull request #9517 from jquick-axway/TIMOB-25376

[TIMOB-25376] Android: Fixed bug where scrolled in ListView rows forget their width/height

hansemannn committed:
[TIMOB-25269] Use final iPhone X launch-image name

hansemannn committed:
Update TiRootViewController.m

hansemannn committed:
Merge branch 'master' into TIMOB-25269-NEW

sgtcoolguy committed:
Fail build/PR via danger.js if no changes to tests, with link to README section on testing. Add dangerfile.js to eslint list

sgtcoolguy committed:
Merge branch 'master' into TIMOB-25269-NEW

Commit:
Merge branch 'master' into TIMOB-25360

hansemannn committed:
[TIMOB-25322] Improve new geo-authentication-flow

Commit:
[MOD-2367] Update archiver to allow handling of java archives

Commit:
[MOD-2367] Compress module java archive

hansemannn committed:
[TIMOB-25322] Provide feedback if user declined

jquick-axway committed:
[6_3_X][TIMOB-25410] Android: Resolve 6.3.0.RC ScrollView regressions.

- [TIMOB-25410] Fixed bug where child views cannot be docked to bottom or right side of empty composite ScrollView content area.
- [TIMOB-25411] Fixed bug where ScrollView events 'click', 'swipe', 'pinch', and 'longclick' no longer fire.
- [TIMOB-25415] Fixed bug where ScrollView.remove() method no longer works.
- [TIMOB-25420] Added 'longclick' event support to ScrollView content area. (Not a regression. Was never supported.)

jquick-axway committed:
Android: Updated API docs for new ScrollView RefreshControl support.

jquick-axway committed:
Android/iOS: Added ScrollView.refreshControl unit test.

jquick-axway committed:
Android/iOS: Fixed typo in "ti.ui.scrollview.test.js" unit test.

maggieaxway committed:
[TIMOB-25423] Android: Ti.UI.Slider on Android 4.x is not working, error introduced on SDK 6.2.x

Update TIUISlider.java

hansemannn committed:
Merge branch 'master' into TIMOB-25423

Commit:
Merge pull request #9513 from jquick-axway/TIMOB-25360

[TIMOB-25360] Android: ScrollView enhancements

jquick-axway committed:
Android: Added new ScrollView "add-insert-remove" unit test.

Commit:
Merge pull request #9545 from jquick-axway/TIMOB-25410

[TIMOB-25410] Android: Resolve 6.3.0.RC ScrollView regressions.

maggieaxway committed:
Merge branch 'master' into TIMOB-25423

maggieaxway committed:
Add Unit Test

Commit:
Merge pull request #9544 from maggieaxway/TIMOB-25423

[TIMOB-25423] Android: Ti.UI.Slider on Android 4.x is not working, er…

jquick-axway committed:
[TIMOB-16257] Android: Fixed crash caused by setting ScrollView.scrollingEnabled on non-UI thread.

- Caused by Android message ID collision with super class' IDs.
- Also resolved potential ID collision with the following features:
  * ActivityIndicator
  * CardView
  * Label
  * Window

jquick-axway committed:
Android/iOS/Windows: Updated "ScrollView.scrollingEnabled" unit test.

ypbnv committed:
Merge branch 'master' into TIMOB-25374

ypbnv committed:
Fix documentation.

ypbnv committed:
Merge branch 'TIMOB-25374' of https://github.com/ypbnv/titanium_mobile into TIMOB-25374

Commit:
[TIMOB-25353] Handle unresolved module dependencies

garymathews committed:
[TIMOB-25266] Fix disableHWAcceleration() for 4.1 (#9507)

Commit:
Merge branch 'master' into TIMOB-16257

Commit:
Merge pull request #9550 from jquick-axway/TIMOB-16257

[TIMOB-16257] Android: Fixed crashed caused by setting ScrollView.scrollingEnabled on non-UI thread

ypbnv committed:
Set default border back to the Proxy when only border color is used.
Add test case for that.

ypbnv committed:
Merge branch 'master' into TIMOB-25374

garymathews committed:
Merge branch 'master' into TIMOB-25051

Commit:
[TIMOB-25363] Fix uri parsing

Commit:
[TIMOB-25441] Support ARM64, update V8 to 6.2.414.36

jquick-axway committed:
Android: Updates Titanium SDK and build tools to compile with Java 7.

- [TIMOB-25442] Updated core Android Titanium SDK code to use Java 7.
- [TIMOB-25448] Updated CLI to build Android modules with Java 7.

ypbnv committed:
Let the system take care of status bar's height compensation.

feons committed:
[TIMOB-24000] Module build should error if the apiversion in manifest does not match the SDKs apiversion

Commit:
[TIMOB-17340] Implement ATTRIBUTE_BASELINE_OFFSET for Android

Commit:
[TIMOB-24679] Remove invalid constant definitions

Commit:
[TIMOB-25363] Fix handling of EXTRA_* URIs

Commit:
[TIMOB-25363] Set type when data is not defined

hansemannn committed:
[TIMOB-24994] Remove Ti.UI.iOS.AdView

hansemannn committed:
[TIMOB-25449] Fix webview spacing

garymathews committed:
Merge branch 'master' into TIMOB-24679

ypbnv committed:
Guard ActionBar reference for undrawn windows.
Add test.

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-25449

ewieberappc committed:
Merge branch 'master' into TIMOB-25300

ewieberappc committed:
Merge branch 'master' into TIMOB-25301

ypbnv committed:
Add a warning.

hansemannn committed:
Merge branch 'master' into TIMOB-25269-NEW

hansemannn committed:
Update _build.js

hansemannn committed:
Update _build.js

hansemannn committed:
[TIMOB-25466] Move iOS 11 guards inside method

# Conflicts:
#	iphone/Classes/TiUIWindowProxy.m

Commit:
[TIMOB-25462] Implement Ti.UI.WebView.userAgent

Commit:
[TIMOB-25462] Add test case

janvennemann committed:
[TIMOB-25059] Add Hyperloop as a pre-packged module

First step of integrating Hyperloop into our SDK by bundling it as a pre-packaged module.

ypbnv committed:
Fix warning message.

Commit:
[TIMOB-25462] Fix test formatting

sgtcoolguy committed:
Auto-add community, android, ios, needs jira, docs, needs cla labels to PRs. Add nice messages to PRs from outside contributors thanking them.

Commit:
Fix ti.cloudpush link

garymathews committed:
Merge pull request #9580 from garymathews/fix_modules

Fix ti.cloudpush link

Commit:
Merge branch 'master' into TIMOB-25212

Commit:
Merge pull request #9383 from garymathews/TIMOB-25212

[TIMOB-25212] Android: Fix strict equality of apiName

hansemannn committed:
Merge branch 'master' into TIMOB-25449

hansemannn committed:
[TIMOB-25459] Change toImage callback-parameter

jawa9000 committed:
Updated CardView.yml per TIDOC-2865 (#9584)

Updated line 21 to include the following:

CardView does not support <Titanium.UI.View.backgroundImage>, <Titanium.UI.View.borderColor>, or <Titanium.UI.View.backgroundGradient>.

mukherjee2 committed:
Merge branch 'master' into TIMOB-25269-NEW

mukherjee2 committed:
Merge branch 'master' into TIMOB-25300

mukherjee2 committed:
Merge branch 'master' into TIMOB-25301

Commit:
[TIMOB-25441] Update libtiverify libraries

hansemannn committed:
[TIMOB-24994] Rephrase error-message

Commit:
[TIMOB-25441] Update module links

hansemannn committed:
Add first bunch of iOS-unit-tests for constants

sgtcoolguy committed:
Merge branch 'master' into TIMOB-24000

sgtcoolguy committed:
Don't publish dependency check results to threadfix server unless it's a master build

sgtcoolguy committed:
Merge branch 'master' into TIMOB-25441

garymathews committed:
[TIMOB-25441] Include ti.playservices

ewieberappc committed:
Merge branch 'master' into TIMOB-25301

ewieberappc committed:
Merge branch 'master' into TIMOB-25300

ewieberappc committed:
Merge branch 'master' into TIMOB-25449

Commit:
Merge pull request #9561 from garymathews/TIMOB-25441

[TIMOB-25441] Android: Support ARM64

Commit:
Merge branch 'master' into TIMOB-25374

Commit:
[TIMOB-25353] Attempt to include missing module dependencies

Commit:
[TIMOB-25353] Fix linting issues

eric34 committed:
Merge pull request #9569 from hansemannn/TIMOB-25449

[TIMOB-25449] iOS 11: Fix WebView spacing

eric34 committed:
Merge pull request #9522 from hansemannn/TIMOB-25300

[TIMOB-25300] iOS 11: Fix title-control in landscape

eric34 committed:
Merge pull request #9533 from hansemannn/TIMOB-25269-NEW

[TIMOB-25269] iOS: Use final iPhone X launch-image names

Commit:
[TIMOB-25487] Prevent kroll from removing null proxy object pointer

vijaysingh-axway committed:
[TIMOB-20404]: iOS ProgressBar gets clipped in parent with height Ti.UI.SIZE (#9273)

* [TIMOB-20404]: iOS ProgressBar gets clipped in parent with height Ti.UI.SIZE

Commit:
[TIMOB-25462] Prevent 'load' event firing twice

feons committed:
[TIMOB-25416] Ambiguous certificates trigger development provisioning… (#9553)

* [TIMOB-25416] Ambiguous certificates trigger development provisioning profiles error

* fix names

* fix style

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-25301

mukherjee2 commented:
@hansemannn @garymathews the status above shows merge conflicts. I'm not sure who is responsible to fix. Can one of you please take a look and resolve?

ypbnv committed:
Merge branch 'master' into TIMOB-25374

garymathews committed:
[TIMOB-25382] Use TiViewProxy for MSG_IDs

Commit:
Merge branch 'master' into TIMOB-24722

Commit:
Merge pull request #9481 from garymathews/TIMOB-24722

[TIMOB-25353] Android: Allow Titanium modules to process module dependencies

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-25301

vijaysingh-axway committed:
[TIMOB-25301] : Window.yml doc updated

ypbnv committed:
Merge branch 'master' into TIMOB-25456

ypbnv committed:
Merge branch 'master' into TIMOB-25374

ypbnv committed:
Merge branch 'master' into TIMOB-25456

mukherjee2 committed:
Merge branch 'master' into TIMOB-25005

Commit:
Merge branch 'master' into TIMOB-25462

mukherjee2 committed:
Merge branch 'master' into TIMOB-25005

hansemannn committed:
[TIMOB-23495] iOS: Add Ti.Filesystem.File "parent" property (parity) (#9285)

* Rework TiFilesystemFile, add parent
* Add unit test
* Deprecate getParent() for parity

* Update ti.filesystem.file.test.js

* Update ti.filesystem.file.test.js

sgtcoolguy committed:
Merge branch 'master' into TIMOB-25462

mukherjee2 committed:
Merge branch 'master' into TIMOB-25301

Commit:
Merge branch 'master' into TIMOB-18500

hansemannn committed:
Merge branch 'master' of github.com:appcelerator/titanium_mobile into TIMOB-10787

# Conflicts:
#	apidoc/Titanium/Platform/Platform.yml
#	iphone/Classes/PlatformModule.h
#	iphone/Classes/PlatformModule.m
#	iphone/Classes/defines.h

hansemannn committed:
Restore API's

hansemannn committed:
Fix linting errors

hansemannn committed:
[TIMOB-10787] Add unit-tests

drauggres committed:
Merge branch 'master' into TIMOB-25051

dieskim committed:
[TIMOB-23645] Android: Added Color to Android notification color behind icon (#9444)

* - Added Color to Android notification to set color of background behind icon via setColor Native Method
- Added Documentation

* Update NotificationProxy.java

* Update Notification.yml

* - added Android >= 5 check

* fixed indentation

* - fixed indentation
- removed spaces
- imported android.os.Build

* added osver: {android: {min: "5.0"}}

sgtcoolguy committed:
Don't do security scans on PRs Takes an additional 5.5 minutes and we don't upload teh results anywhere except on master builds

ypbnv committed:
[TIMOB-24510] Android: Support for custom quick settings tiles. (#9310)

* Support for QuickSettings tiles.

* Add documentation.

* Formatting.
Optimize default label and icon code in build script.

* Fix indentation.

* Fix indentation again.

* Add constants.
Fix parameters.

* Add names.

* Extract dictionary with dialog show params.
Add returns.

* Fix return types.

* More indentation.

* Last attempt.

* Update QuickSettingsService.yml

* Change serviceParser.
Fix default icon code.
Fix parity.html template.

* Docs formatting.

janvennemann committed:
[TIMOB-25059] Update Hyperloop to 3.0.0 Beta 1

hansemannn committed:
[TIMOB-24327] iOS: Add "allowsMultipleSelectionWhenEditing" and "getSelectedRow" in ListView (#9557)

* Enable Multiple Selection in List View

* Added method setAllowsMultipleSelectionDuringEditing
* Added method getSelectedRows
* Added test case

* Update TiUIListView.m

* Update TiUIListViewProxy.m

* Update app.js

* Update ListView.yml

* Update TiUIListViewProxy.m

* Update ListView.yml

* Add unit-tests, fix kroll-thread usage

* [TIMOB-24327] Apply clang-format

* [TIMOB-24327] Update unit-tests from test-suite

* Update ListView.yml

sgtcoolguy committed:
Fixes for Android native module builds using module dependencies

jawa9000 committed:
Updated Animation.yml per TIDOC-1190

Updated the code sample in lines 231-244

hansemannn committed:
Update Window.yml

hansemannn committed:
Merge branch 'master' into TIMOB-25301

mukherjee2 committed:
Merge branch 'master' into TIMOB-25051

janvennemann committed:
Merge branch 'master' into TIMOB-24446

janvennemann committed:
[TIMOB-24446] Fix linting issues

Commit:
[TIMOB-10724] Fix lint issues

mukherjee2 committed:
Merge branch 'master' into TIMOB-24679

Commit:
Merge branch 'master' into TIMOB-24446

Commit:
Merge pull request #9008 from janvennemann/TIMOB-24446

[TIMOB-24446] Allow replacing bundled Android Support Libraries

mukherjee2 committed:
Merge branch 'master' into TIMOB-24679

mukherjee2 committed:
Merge branch 'master' into TIMOB-25301

vijaysingh-axway committed:
Merge branch 'master' into TIMOB-10787

hansemannn committed:
[TIMOB-25504] Fix default-image selection

sgtcoolguy committed:
run git prune on shared reference clone

sgtcoolguy committed:
Merge branch 'master' into TIMOB-10724

mukherjee2 committed:
Merge branch 'master' into TIMOB-25456

mukherjee2 committed:
Merge branch 'master' into TIMOB-24679

garymathews committed:
Merge branch 'master' into TIMOB-25051

garymathews committed:
Merge branch 'master' into TIMOB-10724

ewieberappc committed:
Merge branch 'master' into TIMOB-10787

mukherjee2 committed:
Merge branch 'master' into TIMOB-24000

hansemannn committed:
Merge branch 'master' into TIMOB-25005

hansemannn committed:
Merge branch 'master' into TIMOB-25005

sgtcoolguy committed:
Merge branch 'master' into TIMOB-25462

Commit:
Merge pull request #9265 from garymathews/TIMOB-10724

[TIMOB-10724] Android: Remove native module gperf dependency

mukherjee2 committed:
Merge pull request #9563 from feons/TIMOB-24000

[TIMOB-24000] Module build should error if the apiversion in manifest…

eric34 committed:
Merge pull request #8953 from hansemannn/TIMOB-10787

[TIMOB-10787] iOS: Expose modern UUID-replacements

mukherjee2 committed:
Merge pull request #9570 from ypbnv/TIMOB-25456

[TIMOB-25456] Android: Crash after calling JSON.stringify on Ti.UI.Window

jquick-axway committed:
Android: Lowered min "targetSdkVersion" from 25 to 23.

Commit:
Merge pull request #9595 from jquick-axway/TIMOB-25510

Android: Lowered min "targetSdkVersion" from 25 to 23.

eric34 committed:
Merge pull request #9576 from garymathews/TIMOB-25462

[TIMOB-25462] Android: Implement Ti.UI.WebView.userAgent

eric34 committed:
Merge pull request #9476 from maggieaxway/TIMOB-18500

[TIMOB-18500] Android: event.cancel not set properly for optionsDialog

eric34 committed:
Merge pull request #9510 from ypbnv/TIMOB-25374

[TIMOB-25374] Android: borderColor not visible on Android tiSDK 6.2.2

eric34 committed:
Merge pull request #9564 from garymathews/TIMOB-17340

[TIMOB-17340] Android: Implement Ti.UI.ATTRIBUTE_BASELINE_OFFSET

eric34 committed:
Merge pull request #9239 from hansemannn/TIMOB-25005

[TIMOB-25005] Remove Ti.UI.iPhone namespace

eric34 committed:
Merge pull request #9565 from garymathews/TIMOB-24679

[TIMOB-24679] Android: Remove invalid constant definitions

eric34 committed:
Merge pull request #9276 from drauggres/TIMOB-25051

[TIMOB-25051] fix persistentObjects leak

eric34 committed:
Merge pull request #9475 from vijaysingh-axway/TIMOB-25301

[TIMOB-25301]:iPhone X: Need to be able to control the Insets / Layout Margins

jquick-axway commented on android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java:
```diff
@@ -676,6 +676,18 @@ public void scrollToBottom()
 
+	public void scrollToTop()
+	{
+		View view = getNativeView();
```
Once we update this PR with what's on master, we'll need to change this line of code...

```java
View view = getNativeView();
```

...to...

```java
View view = this.scrollView;
```

This is because we've added `RefreshControl` support to scroll views as of 6.3.0 and the `getNativeView()` method will return a `SwipeRefreshLayout` object instead.

jquick-axway committed:
Merged

jquick-axway committed:
Android/iOS/Windows: Added unit test for ScrollView.scrollToTop() function.

m1ga commented:
@jquick-axway you already change the `getNativeView()`, right?

jquick-axway commented:
> you already changed the getNativeView(), right?

@m1ga, I have. The goal was to get this PR into 7.0.0 since the Windows PR has already been merged in. But when I resolved the conflicts, this PR is now showing the entire commit history of the merge. Hmm...

jquick-axway commented:
@m1ga, since I screwed up the commit history for this PR, I've set up a new PR #9601 to replace this one. Your commit changes are still intact and credit was given where credit was due.

I've also discovered that animated scrolling in a vertical ScrollView via `scrollTo()` and your new `scrollToTop()` method no longer worked on Android as of Titanium 6.2.2. This is due to us using Google's `NestedScrollView` in place of the `ScrollView` Java class. It appears to be a bug in the `NestedScrollView` class. The new PR resolves this issue by disabling animated scrolling in this case (kind of a bummer, but haven't found a better work-around).

jquick-axway commented:
Closing in favor of PR #9601.
